core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -51,6 +179,18 @@ public synchronized void performCleanup() {
         registeredNetworks.forEach(this::removeNetwork);
     }
 
+    /**
+     * Register a filter to be cleaned up.
+     *
+     * @param filter the filter
+     */
+    public void registerFilterForCleanup(List<Map.Entry<String, String>> filter) {
+        synchronized (REGISTERED_FILTERS) {
+            REGISTERED_FILTERS.add(filter);
+            REGISTERED_FILTERS.notifyAll();
+        }
+    }
+
     /**
      * Register a container to be cleaned up, either on explicit call to stopAndRemoveContainer, or at JVM shutdown.
```

Comment:
Shall we remove this method now?

Reply:
No, it's public. We can not remove it now